src/gnatss/main.py

```diff
@@ -781,27 +782,55 @@ def main(
         Extracted distance from center as dataframe, by default None
     process_ds : Union[xr.Dataset, None]
         Extracted process results as xarray dataset, by default None
+    outliers_df : Union[pd.DataFrame, None]
+        Extracted residual outliers as dataframe, by default None
     """
     all_observations = load_data(all_files_dict, config)
 
     # Extracts distance from center when specified
     dist_center_df = None
     if extract_dist_center:
         dist_center_df = extract_distance_from_center(all_observations, config)
-
+        typer.echo("Filtering out data outside of distance limit...")
+        # Filter out data that is outside of the distance limit
+        all_observations = all_observations[
```

Comment:
Should we break this dataframe operation into multiple lines like below -

```
# Extract distance limit
distance_limit = config.solver.distance_limit

# Extract the rows of observations with distances beyond the limit
filtered_rows = dist_center_df[dist_center_df[constants.GPS_DISTANCE] > distance_limit][constants.garpos.ST]

# Filter out data based on the filtered rows and reset index
all_observations = all_observations[~all_observations[constants.garpos.ST].isin(filtered_rows)].reset_index(drop=True)
```




# Extract the IDs of observations with distances beyond the limit
filtered_ids = dist_center_df[dist_center_df[DISTANCE_COLUMN] > distance_limit][DIST_COLUMN]

# Filter out data based on the filtered IDs and reset index
filtered_observations = all_observations[~all_observations[DIST_COLUMN].isin(filtered_ids)].reset_index(drop=True)